src/Symfony/Component/Mailer/Bridge/Amazon/Transport/SesTransportFactory.php

```diff
@@ -11,35 +11,65 @@
 
 namespace Symfony\Component\Mailer\Bridge\Amazon\Transport;
 
+use AsyncAws\Core\Configuration;
+use AsyncAws\Ses\SesClient;
+use Symfony\Component\HttpClient\HttpClient;
 use Symfony\Component\Mailer\Exception\UnsupportedSchemeException;
 use Symfony\Component\Mailer\Transport\AbstractTransportFactory;
 use Symfony\Component\Mailer\Transport\Dsn;
 use Symfony\Component\Mailer\Transport\TransportInterface;
 
 /**
  * @author Konstantin Myakshin <molodchick@gmail.com>
+ * @author Jérémy Derussé <jeremy@derusse.com>
  */
 final class SesTransportFactory extends AbstractTransportFactory
 {
     public function create(Dsn $dsn): TransportInterface
     {
         $scheme = $dsn->getScheme();
-        $user = $this->getUser($dsn);
-        $password = $this->getPassword($dsn);
         $region = $dsn->getOption('region');
-        $host = 'default' === $dsn->getHost() ? null : $dsn->getHost();
-        $port = $dsn->getPort();
 
-        if ('ses+api' === $scheme) {
-            return (new SesApiTransport($user, $password, $region, $this->client, $this->dispatcher, $this->logger))->setHost($host)->setPort($port);
+        if ('ses+smtp' === $scheme || 'ses+smtps' === $scheme) {
+            return new SesSmtpTransport($this->getUser($dsn), $this->getPassword($dsn), $region, $this->dispatcher, $this->logger);
         }
 
-        if ('ses+https' === $scheme || 'ses' === $scheme) {
-            return (new SesHttpTransport($user, $password, $region, $this->client, $this->dispatcher, $this->logger))->setHost($host)->setPort($port);
-        }
+        if (!class_exists(SesClient::class)) {
+            if (!class_exists(HttpClient::class)) {
+                throw new \LogicException(sprintf('You cannot use "%s" as the HttpClient component or AsyncAws package is not installed. Try running "composer require async-aws/ses".', __CLASS__));
+            }
 
-        if ('ses+smtp' === $scheme || 'ses+smtps' === $scheme) {
-            return new SesSmtpTransport($user, $password, $region, $this->dispatcher, $this->logger);
+            trigger_deprecation('symfony/amazon-mailer', '5.1', 'Using the "%s" transport without AsyncAws is deprecated, use "new %s()" instead.', __METHOD__, \get_called_class());
```

Comment:
`use new SesTransportFactory()` looks wrong to me in the message.

Reply:
thanks, reworded